handlers/related_data.go

```diff
@@ -0,0 +1,133 @@
+package handlers
+
+import (
+	"context"
+	"net/http"
+	"path"
+	"sync"
+
+	zebedeeCli "github.com/ONSdigital/dp-api-clients-go/v2/zebedee"
+	"github.com/ONSdigital/dp-frontend-search-controller/cache"
+	"github.com/ONSdigital/dp-frontend-search-controller/config"
+	"github.com/ONSdigital/dp-frontend-search-controller/data"
+	"github.com/ONSdigital/dp-frontend-search-controller/mapper"
+	core "github.com/ONSdigital/dp-renderer/v2/model"
+	searchAPI "github.com/ONSdigital/dp-search-api/api"
+	searchModels "github.com/ONSdigital/dp-search-api/models"
+	searchSDK "github.com/ONSdigital/dp-search-api/sdk"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// list of content types that have /previousreleases
+var knownRelatedDataTypesTypes = []string{
+	"bulletin",
+	"article",
+	"compendium_landing_page",
+}
+
+func readRelatedData(w http.ResponseWriter, req *http.Request, cfg *config.Config, zc ZebedeeClient, rend RenderClient, searchC SearchClient,
+	accessToken, collectionID, lang string, cacheList cache.List,
+) {
+	ctx, cancel := context.WithCancel(req.Context())
+	defer cancel()
+	var err error
+	template := "related-list-pages"
+	urlPath := path.Dir(req.URL.Path)
+	urlQuery := req.URL.Query()
+
+	sanitisedParams := sanitiseQueryParams(allowedPreviousReleasesQueryParams, urlQuery)
+
+	// check page type
+	pageData, err := checkAllowedPageTypes(ctx, w, zc, accessToken, collectionID, lang, urlPath, knownRelatedDataTypesTypes)
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	// get cached navigation data
+	navigationCache := getNavigationCache(ctx, w, req, cacheList, lang)
+
+	validatedQueryParams, validationErrs := data.ReviewPreviousReleasesQueryWithParams(ctx, cfg, sanitisedParams, urlPath)
```

Comment:
why 'previousreleasesquery' here? either this should be more generic or have it's own?

Reply:
Given there is no user interaction here, do we need to do this?

Reply:
Strike that - there's pagination.

Reply:
Uhh yeah, it's used on both. We could potentially rename it to some more generic so works in both scenarios?

Reply:
Can defer this to refactoring work.